python/src/main/python/yaml-template/main.py

```diff
@@ -21,26 +21,10 @@
 from apache_beam.yaml import cache_provider_artifacts
 
 
-def _pipeline_spec_file_from_args(known_args):
-    if known_args.yaml:
-        return known_args.yaml
-    else:
-        raise ValueError(
-            "--yaml must be set.")
-
-
 def run(argv=None):
-    parser = argparse.ArgumentParser()
-    parser.add_argument(
-        '--yaml',
-        dest='yaml',
-        help='Input YAML file in Cloud Storage.')
-    known_args, pipeline_args = parser.parse_known_args(argv)
-    pipeline_spec_file = _pipeline_spec_file_from_args(known_args)
+    _, args = argparse.ArgumentParser().parse_known_args(argv)
     cache_provider_artifacts.cache_provider_artifacts()
-
-    main.run(argv=pipeline_args + [f"--pipeline_spec_file={pipeline_spec_file}",
-                                   "--save_main_session"])
+    main.run(argv=args + ["--save_main_session"])
```

Comment:
Why do we want save main session? (Generally it's best to avoid if we can.)

Reply:
I'm honestly not sure, I just noticed the other python templates use it. What is the reason to avoid it? Is there ever a reason to use it?

Reply:
Removed.